dashboard/src/pages/Dashboard/Dataset/DatasetEvents.tsx

```diff
@@ -78,70 +80,81 @@ export const DatasetEvents = () => {
+                          <th
+                            scope="col"
+                            class="py-3.5 pl-4 pr-3 text-left text-sm font-semibold text-gray-900 sm:pl-3"
+                          >
+                            Event Type
+                          </th>
+                          <th
+                            scope="col"
+                            class="px-3 py-3.5 text-left text-sm font-semibold text-gray-900"
+                          >
+                            Time
+                          </th>
+                          <th
+                            scope="col"
+                            class="px-3 py-3.5 text-left text-sm font-semibold text-gray-900"
+                          >
+                            Metadata
+                          </th>
+                        </tr>
+                      </thead>
+                      <tbody class="bg-white">
+                        <For each={events()}>
+                          {(event) => (
+                            <tr class="even:bg-gray-50">
+                              <td
+                                classList={{
+                                  "whitespace-nowrap py-4 pl-4 pr-3 text-sm font-medium sm:pl-3":
+                                    true,
+                                  "text-gray-900":
+                                    !event.event_type.includes("failed"),
+                                  "text-red-500":
+                                    event.event_type.includes("failed"),
+                                }}
+                              >
+                                {event.event_type.includes("failed")
+                                  ? "ERROR"
+                                  : "INFO"}
+                              </td>
+                              <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-500 sm:pl-3">
+                                {event.event_type}
+                              </td>
+                              <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-500">
+                                {event.created_at}
+                              </td>
+                              <td class="text-wrap max-w-md overflow-ellipsis break-words px-3 py-4 text-sm text-gray-500">
+                                {JSON.stringify(event.event_data)}
+                              </td>
+                            </tr>
+                          )}
+                        </For>
+                      </tbody>
+                    </table>
+                  </Show>
                 </div>
               </div>
             </div>
+            <Show when={loading()}>
+              <div class="flex items-center justify-center">
+                <img
+                  src="https://media.tenor.com/17U_UB9LwVQAAAAi/loading-response.gif"
+                  alt="loading spinner"
+                  class="h-12 w-12"
+                />
```

Comment:
Just make this an animate-spin with `rounded-full` class on a div. Not a fan of having this URL